### PR TITLE
Replayer cron job for mainnet

### LINKED ISSUE #12993: Replayer cron job
Run a daily cron job that replays a day's worth of blocks on mainnet

Use checkpoint file from previous day.

If the replaying is successful, we can feel confident about migrating data for the hard fork.

### PR DESCRIPTION
**Pull request by psteckler**
Cron job to run replayer on latest mainnet archive dump. Runs 2 hours after the day's dump.

Uploads a checkpoint file to gs://mina-replayer-checkpoints. If there are any errors, also uploads the replayer log. Until we fix the bad nonce issue, there will be errors.

Tested by running the cron job and viewing the logs. The only errors were the expected nonce errors. The error file was not uploaded, because there was an earlier error file with the same name that could not be overwritten. I've since deleted that earlier error file.

Closes #12993.

### CODE REVIEW

**Comment by psteckler** [comment]
!ci-build-me

**Comment by psteckler** [comment]
Note: I need to update the image used to contain the fix in #13460.

**Comment by psteckler** [comment]
!ci-build-me

**Review by QuiteStochastic — APPROVED** [review]

File: helm/cron_jobs/mainnet-replayer-cronjob.yaml [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+               curl https://packages.cloud.google.com/apt/doc/apt-key.gpg | sudo apt-key add - ;
+               apt-get update && apt-get install -y google-cloud-cli ;
+               ARCHIVE_DUMP_URI=$(gsutil ls gs://mina-archive-dumps/mainnet-archive-dump-*.sql.tar.gz | sort -r | head -n 1);
+               ARCHIVE_DUMP=$(basename $ARCHIVE_DUMP_URI);
+               ARCHIVE_SQL=$(basename $ARCHIVE_DUMP_URI .tar.gz);
+               echo "Getting archive dump" $ARCHIVE_DUMP_URI;
+               gsutil -o Credentials:gs_service_key_file=/gcloud/keyfile.json cp $ARCHIVE_DUMP_URI . ;
+               MOST_RECENT_CHECKPOINT_URI=$(gsutil ls gs://mainnet-replayer-checkpoints/replayer-checkpoint-*.json | sort -r | head -n 1);
+               MOST_RECENT_CHECKPOINT=$(basename $MOST_RECENT_CHECKPOINT_URI);
+               echo "Getting replayer checkpoint file" $CHECKPOINT_FILE;
```
[the remark this answers is not in the document]
Comment (psteckler) [reply helm/cron_jobs/mainnet-replayer-cronjob.yaml:40]: well-spotted, thanks

File: helm/cron_jobs/mainnet-replayer-cronjob.yaml [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+#   we reverse-sort the filenames, take the most recent one
+# - ALTER USER takes a password bracketed in single-quotes or $$s
+#   too hard to make the single-quotes work inside single-quotes, so use $$
+#   the space in the password makes sure a $ isn't applied to the following string
+apiVersion: batch/v1beta1
+kind: CronJob
+metadata:
+  name: mainnet-replayer-cronjob
+spec:
+  concurrencyPolicy: Replace
```
[the remark this answers is not in the document]
Comment (psteckler) [reply helm/cron_jobs/mainnet-replayer-cronjob.yaml:14]: > Is this desired?

It shouldn't need more than an hour. I was copying from the other cron jobs. Should those other jobs also have `Forbid`?

[the remark this answers is not in the document]
Comment (psteckler) [reply helm/cron_jobs/mainnet-replayer-cronjob.yaml]: I made that change, doesn't hurt.

**Comment by deepthiskumar** [comment]
!approved-for-mainnet

**Comment by mrmr1993** [comment]
!ci-build-me

**Comment by psteckler** [comment]
!ci-build-me

**Pull request merged**